src/StructuredLogViewer/Controls/BuildControl.xaml.cs

```diff
@@ -1024,11 +1030,30 @@ private object FindInFiles(string searchText, int maxResults, CancellationToken
                     return null;
                 }
 
-                var haystack = file.Value;
-                var resultsInFile = haystack.Find(searchText);
-                if (resultsInFile.Count > 0)
+                if (!string.IsNullOrEmpty(searchText) && searchText.StartsWith("$secret"))
                 {
-                    results.Add((file.Key, resultsInFile.Select(lineNumber => (lineNumber, haystack.GetLineText(lineNumber)))));
+                    var word = searchText.Replace("$secret", string.Empty).Trim();

```

Comment:
instead of manually parsing the query, try just doing `var matcher = new NodeQueryMatcher(searchText)`

the resulting matcher will have the `TypeKeyword`, the `Terms` and the `NotMatchers`.

Reply:
also keep in mind that you can have multiple not matchers, so you may want to pass multiple to SearchSecrets()

Reply:
oh, and parsing the searchText can also be done outside the foreach loop, to avoid repetitive work